### PR TITLE
[WIP] Component function

### PR DESCRIPTION
**Pull request by AndrewZamora**
I managed to get the form to work after submitting it. I know there is a lot of prop drilling but I hope it makes the handleSubmit function somewhat reusable. Is this a step in the right direction in solving issue #2 ?

### CODE REVIEW

File: src/components/Login.js [hunk 1]
```diff
@@ -9,35 +9,32 @@ class Login extends Component {
...
     return (
       <div className="container">
         <h3>Login to...</h3>
         <div className="row">
-          <form className="col s12" onSubmit={event => this.handleSubmit(event)}>
+          <form className="col s12" onSubmit={(event) => this.props.onSubmit(event,this.state)}>
             {/* ---password--- */}
             <div className="row">
               <div className="input-field col s12">
-                <input id="password" type="password" name='password' className="validate" autocomplete="off"
```
Comment (martinheidegger) [inline src/components/Login.js]: While its generally a good idea to catch errors, it would be good to keep it in different PR's ;)

File: src/components/Login.js [hunk 2]
```diff
@@ -9,35 +9,32 @@ class Login extends Component {
...
-    event.preventDefault();
-  }
+
   render() {
     return (
       <div className="container">
         <h3>Login to...</h3>
         <div className="row">
-          <form className="col s12" onSubmit={event => this.handleSubmit(event)}>
+          <form className="col s12" onSubmit={(event) => this.props.onSubmit(event,this.state)}>
```
Comment (martinheidegger) [inline src/components/Login.js]: By passing it on like this, the enclosing Component needs to know that it needs to do the `event.preventDefault()`. It would be better to keep this inside the the Login Form.

File: src/App.js [hunk 3]
```diff
@@ -4,12 +4,23 @@ import './App.css';
 import Navbar from './components/Navbar';
 
 class App extends Component {
+    constructor(props) {
+        super(props);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleSubmit = (event, submitInfo) => {
```
Comment (martinheidegger) [inline src/App.js]: The "handleSubmit" of the app should be called "handleLogin", not?

File: src/components/Login.js (hunk 1 above)
Comment (AndrewZamora) [reply src/components/Login.js]: OK I will set it back and make a separate PR for it.

File: src/components/Login.js [hunk 4]
```diff
@@ -9,35 +9,36 @@ class Login extends Component {
...
+    this.props.onSubmit(this.state);
   }
+
   render() {
     return (
       <div className="container">
         <h3>Login to...</h3>
         <div className="row">
-          <form className="col s12" onSubmit={event => this.handleSubmit(event)}>
+          <form className="col s12" onSubmit={(event) => this.handleSubmit(event)}>
```
Comment (martinheidegger) [inline src/components/Login.js]: I am not quite sure why you added the `()`  braces.

**Comment by martinheidegger** [comment]
LGTM

File: src/components/Login.js (hunk 4 above)
Comment (AndrewZamora) [reply src/components/Login.js]: 🤦‍♂️ I'm not sure why I did that either. I will remove them and merge this branch.
